enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,522 @@
+Any specific actions are related to the delegate CNIs. For example: creating veth pair,
+attaching one end of veth to the bridge and making the other end available within the Pod is
+responsibility of `bridge` CNI is this example.
+MicroShift team will create tests for CNIs that will be declared as supported. However these tests
+will not explore the breadth and depth of possible network setups, so ultimately the responsibility
+for correctness of the configuration is up to the user.
+
+#### Manifests
+
+MicroShift will provide Multus based on thin architecture because of the resource consumption
+(see alternatives for more information).
+Because of the differences with OpenShift (which uses thick Multus plugin), existing OpenShift
+manifests will require changes to make them suitable for MicroShift. These manifests will reside
+in MicroShift repository.
+
+Because one of the required IPAMs is the `dhcp` (dynamic), manifests will also include
+a DHCP server DaemonSet.
+
+Updating necessary image references will be part of existing rebase procedure.
+
+#### RPM package
+
+RPM spec to build `microshift-multus` and `microshift-multus-release-info` RPMs will be part of
+existing `microshift.spec` file.
+The RPM will include:
+- manifests required to deploy Multus on MicroShift
+- CRI-O drop-in config to use Multus instead of ovn-kubernetes (which will require reorganization
+  of currently existing MicroShift's CRI-O configs)
+- greenboot healthcheck script
+- cleanup script plugin
+
+#### Container images: Multus and network plugins
+
+Because the Multus image used by OpenShift has a large size that is not acceptable for edge devices, new image
+will be prepared and will only include relevant artifacts such as entrypoint script/binary and
+Multus CNI binary (which is copied to host's `/opt/cni/bin`).
+
+To supply network plugins (CNIs) such as `bridge`, `ipvlan`, and `macvlan` a new image will be prepared
+so MicroShift uses the same binaries as OpenShift (alternative is using RHEL's networkplugins RPM - see alternatives).
+This image will also contain IPAM binaries such as `static`, `dynamic (DHCP)`, and `host-local`.
+
+Both of these images will be part of the OpenShift payload which MicroShift references during rebase procedure.
+
+#### Hypershift [optional]
+
+No, enhancement is MicroShift specific.
+
+### Risks and Mitigations
+
+There may be a race condition between Multus on MicroShift and other services on the host that also
+configure the host's networking
+Taking `bridge` CNI as an example: when the bridge interface does not exist, it will be created
+when a Pod requiring that interface is created.
+If user expects something else will create the interface, they will need to configure system to start
+MicroShift after other services.
+
+Multus using thin plugin architecture creates a kubeconfig when the DaemonSet starts and copies it
+out of Pod to `/opt/cni/bin` on the host filesystem so the Multus CNI binary can use it to get
```

Comment:
'/opt/cni/bin' is for cni binary path, not kubeconfig.
I forgot which path does multus place the kubeconfig.

Reply:
You're right, fixed it